src/blocks/Transaction.vue

```diff
@@ -507,50 +507,55 @@ export default Vue.extend({
       this.loading = false;
     },
     async withdraw(): Promise<void> {
-      const syncProvider = walletData.get().syncProvider;
       const txAmount = utils.parseToken((this.chosenToken as ZkInBalance).symbol, this.inputtedAmount);
       this.tip = "Confirm the transaction to withdraw";
       if (this.feesObj === undefined) {
-        throw new Error("Fee fetching error :(");
+        throw new Error("Fee fetching error");
       }
-      const withdrawTransaction = await withdraw({
+      const withdrawTransactions = await withdraw({
         address: this.inputtedAddress,
         token: (this.chosenToken as ZkInBalance).symbol,
         feeToken: this.feeToken.symbol,
         amount: txAmount.toString(),
         fastWithdraw: this.transactionMode === "fast",
-        fees: (this.transactionMode === "fast" ? this.feesObj?.fast : this.feesObj?.normal) as string,
+        fee: (this.transactionMode === "fast" ? this.feesObj?.fast : this.feesObj?.normal) as string,
         store: this.$accessor,
+        accountActivationFee: this.activateAccountFee,
       });
 
-      let receipt: TransactionReceipt;
       this.transactionInfo.amount!.amount = txAmount.toString();
       this.transactionInfo.amount!.token = this.chosenToken as ZkInBalance;
       this.transactionInfo.fee!.token = this.feeToken;
-      if (!Array.isArray(withdrawTransaction)) {
-        this.transactionInfo.hash = withdrawTransaction.txHash;
-        this.transactionInfo.explorerLink = APP_ZKSYNC_BLOCK_EXPLORER + "/transactions/" + withdrawTransaction.txHash;
-        this.transactionInfo.fee!.amount = withdrawTransaction.txData.tx.fee;
-        this.transactionInfo.recipient = {
-          address: withdrawTransaction.txData.tx.to,
-          name: this.chosenContact ? this.chosenContact.name : "",
-        };
-        this.tip = "Waiting for the transaction to be mined...";
-        receipt = await withdrawTransaction?.awaitReceipt();
-      } else {
-        this.transactionInfo.hash = withdrawTransaction[0].txHash;
-        this.transactionInfo.explorerLink = APP_ZKSYNC_BLOCK_EXPLORER + "/transactions/" + withdrawTransaction[0].txHash;
-        this.transactionInfo.fee!.amount = withdrawTransaction[1].txData.tx.fee;
-        this.transactionInfo.recipient = {
-          address: withdrawTransaction[0].txData.tx.to,
-          name: this.chosenContact ? this.chosenContact.name : "",
-        };
-        this.tip = "Waiting for the transaction to be mined...";
-        receipt = await syncProvider!.notifyTransaction(withdrawTransaction[0].txHash, "COMMIT");
+
+      const activateTransaction = this.$accessor.wallet.isAccountLocked ? withdrawTransactions.shift() : undefined;
+      const withdrawTransaction = withdrawTransactions.shift();
```

Comment:
Maybe add a little more explanation about what is done here? It is someone hard to follow what all those `shifts` do. So that the future programmers would not need to remember what the first elements are and why we delete them